src/lib/middleware/validator.js

```diff
@@ -0,0 +1,54 @@
+export const categoryCreateValidator = store => next => action => {
+  const isCategory = action.type && action.type.startsWith('CATEGORY');
+
+  if (isCategory) {
+    try {
+      const category = action.payload;
+      let notValid = !category.title || !category.budget;
+
+      if (notValid) {
+        throw new Error('VALIDATION ERROR: category must include title and budget');
+      }
+
+      notValid = category.budget && category.budget < 0;
+      if (notValid) {
+        throw new Error('VALIDATION ERROR: budget must be a value greater than 0');
+      }
+
+      return next(action);
+    } catch (e) {
+      e.action = action;
+      throw e;
+    }
+  } else {
+    return next(action);
+  }
+};
+
+export const expenseCreateValidator = store => next => action => {
```

Comment:
Looks like you forgot to check for input. At this point you can create an Expense with a blank 'expense' and blank 'cost' field, but your over budget validation works.